internal/app/synchronizer/evaluator_client.go

```diff
@@ -31,93 +32,74 @@ import (
 	"open-match.dev/open-match/pkg/pb"
 )
 
-type evaluator interface {
-	evaluate(context.Context, []*pb.Match) ([]*pb.Match, error)
-}
-
-type clientType int
-
-// Different type of clients supported for evaluation.
-const (
-	none clientType = iota
-	grpcEvaluator
-	httpEvaluator
-)
-
 var (
 	evaluatorClientLogger = logrus.WithFields(logrus.Fields{
 		"app":       "openmatch",
 		"component": "app.synchronizer.evaluator_client",
 	})
 )
 
-type evaluatorClient struct {
-	clType     clientType
-	cfg        config.View
-	grpcClient pb.EvaluatorClient
-	httpClient *http.Client
-	baseURL    string
-	m          sync.Mutex
+type evaluator interface {
+	evaluate(context.Context, []*pb.Match) ([]*pb.Match, error)
 }
 
-// evaluate method triggers execution of user evaluation function. It initializes the configured
-// GRPC or HTTP client. If the client was already initialized, initialize is a no-op.
-func (ec *evaluatorClient) evaluate(ctx context.Context, proposals []*pb.Match) (result []*pb.Match, err error) {
-	if err := ec.initialize(); err != nil {
-		return nil, err
+var errNoEvaluatorType = errors.New("unable to determine evaluator type, either api.evaluator.grpcport or api.evaluator.httpport must be specified in the config")
+
+func newEvaluator(cfg config.View) evaluator {
+	newInstance := func(cfg config.View) (interface{}, error) {
+		if cfg.IsSet("api.evaluator.grpcport") {
+			return newGrpcEvaluator(cfg)
+		}
+		if cfg.IsSet("api.evaluator.httpport") {
+			return newHTTPEvaluator(cfg)
+		}
+		return nil, errNoEvaluatorType
```

Comment:
Will this surface to the caller as InternalError? We should a grpc error code here? Unavailable? invalid argument?

Reply:
Good catch!  Changed to FailedPrecondition.